hermes/bin/src/state.rs

```diff
@@ -60,3 +62,32 @@ impl Context {
         self.counter
     }
 }
+
+/// All Hermes runtime extensions states need to implement this.
+pub(crate) trait Stateful {
+    /// Initial state for the given context
+    fn new(ctx: &Context) -> Self;
+}
+
```

Comment:
Move the above code to a `runtime/extensions/state.rs` module.
Keep everything below this line here.

Reply:
done